### PR TITLE
Updated DNSSEC05

### PR DESCRIPTION
**Pull request by matsduf**
* Solves issue #763. 
* Meet specification in new RFC 8624 with more strict requirements.
* Decreased wordiness.
* Increased default level for several messages.
* Message ALGORITHM_DELETE_DS has been removed (algorithm handled
  as ALGORITHM_NOT_ZONE_SIGN).
* Message ALGORITHM_INDIRECT_KEY has been removed (algorithm handled
  as ALGORITHM_RESERVED).
* Some editorial updates.

When this PR has been merged, an issue should be created in Zonemaster-Engine.

### CODE REVIEW

**Comment by sandoche2k** [comment]
@sandoche2k @mattias-p to do the review.

File: docs/specifications/tests/DNSSEC-TP/dnssec05.md [hunk 1]
```diff
@@ -5,20 +5,24 @@
...
 ## Objective
 
-A domain name (zone) should only use DNSKEY algorithms that are specified 
-by IANA to be used for a DNSKEY. A public domain name (zone) should not
-use private algorithms. The IANA registry of [DNSSEC Algorithm Numbers]
-specifies which algorithms to use.
+A domain name (zone) should only use DNSKEY algorithms that are specified
+by [RFC 8624], section 3.1 and the [IANA registry] of *DNSSEC Algorithm
+Numbers* to be used for DNSSEC signing. A public domain name (zone) should not use
+private algorithms.
```
Comment (sandoche2k) [inline docs/specifications/tests/DNSSEC-TP/dnssec05.md]: I am not sure is it a good idea to have both the RFC and the registry as reference. The registry will be updated, whereas the RFC will not be..

File: docs/specifications/tests/DNSSEC-TP/dnssec05.md [hunk 2]
```diff
@@ -27,29 +31,25 @@ The DNSKEY record is defined in [RFC 4034, section 2].
...
+      2. If the algorithm is reserved (algorithm
+         4, 9, 11, 123-252 or 255), output *[ALGORITHM_RESERVED]*.
+      3. If the algorithm is unassigned (algorithm
+         17-122), output *[ALGORITHM_UNASSIGNED]*.
+      4. If the algorithm is private algorithm
+         (algorithm 253-254), output *[ALGORITHM_PRIVATE]*.
+      5. If the algorithm is not meant for zone signing (algorithm
+         0-2), output *[ALGORITHM_NOT_ZONE_SIGN]*.
+      6. If the algorithm is not rekommended for zone signing (algorithm
+         5, 7 or 10), output *[ALGORITHM_NOT_RECOMMENDED]*.
```
Comment (sandoche2k) [inline docs/specifications/tests/DNSSEC-TP/dnssec05.md]: Why did you remove INDIRECT_KEY?

File: docs/specifications/tests/DNSSEC-TP/dnssec05.md (hunk 1 above)
Comment (matsduf) [reply docs/specifications/tests/DNSSEC-TP/dnssec05.md]: I agree that it is complex, but I think we have to meet the stricter specification in the RFC at the same time as we should keep a reference to the registry. There algorithms that the registry covers that are not mentioned in the RFC.

When the registry is updated, there will be a simple update if they just adapt to the RFC (which I think they will) or else we have to decide how we should balance.

Independent of what we do, we have to follow the change of RFC and the update of the registry.

File: docs/specifications/tests/DNSSEC-TP/dnssec05.md (hunk 2 above)
Comment (matsduf) [reply docs/specifications/tests/DNSSEC-TP/dnssec05.md]: I did not remove it. It is treated as "reserved" instead. It is quite unclear what it really is. I guess it is better to move it back.
